src/Common/src/Interop/Unix/libssl/Interop.OpenSsl.cs

```diff
@@ -118,23 +118,21 @@ internal static int Encrypt(SafeSslHandle context, IntPtr buffer, int offset, in
                         break;
 
                     default:
-                        throw CreateSslException("OpenSsl::Encrypt failed");
+                        throw CreateSslException(SR.net_ssl_encrypt_failed, errorCode);
                 }
             }
             else
             {
-                int capacityNeeded = libssl.BIO_ctrl_pending(context.OutputBio);      
+                int capacityNeeded = libssl.BIO_ctrl_pending(context.OutputBio);
 
-                if (capacityNeeded > bufferCapacity)
-                {
-                    throw CreateSslException("OpenSsl::Encrypt capacity needed is more than buffer capacity. capacityNeeded = " + capacityNeeded + "," + "bufferCapacity = " + bufferCapacity);
-                }             
+                Debug.Assert(bufferCapacity >= capacityNeeded, "Input buffer of size " + bufferCapacity +
+                                                              " bytes is insufficient since encryption needs " + capacityNeeded + " bytes.");
```

Comment:
Asserts should only be used if the situation can never occur in practice.  Does BIO_ctrl_pending always return an appropriate value here?